internal/gatewayapi/translator.go

```diff
@@ -1226,7 +1226,7 @@ func irListenerName(listener *ListenerContext) string {
 }
 
 func irInfraPortName(listener *ListenerContext) string {
-	return fmt.Sprintf("%s-%s", listener.gateway.Namespace, listener.gateway.Name)
+	return fmt.Sprintf("%s-%s-%s", listener.gateway.Namespace, listener.gateway.Name, listener.Name)
```

Comment:
is `irInfraPortName` needed anymore ? can we use `listener.Name` directly since the infra service will be unique so there shouldnt be any conflicts afaik